pkg/providers/instancetype/instancetype.go

```diff
@@ -181,34 +139,49 @@ func (p *DefaultProvider) List(ctx context.Context, kc *v1.KubeletConfiguration,
 			instanceTypeLabel: *i.InstanceType,
 		}).Set(float64(aws.Int64Value(i.MemoryInfo.SizeInMiB) * 1024 * 1024))
 
-		// !!! Important !!!
-		// Any changes to the values passed into the NewInstanceType method will require making updates to the cache key
-		// so that Karpenter is able to cache the set of InstanceTypes based on values that alter the set of instance types
-		// !!! Important !!!
-		return p.instanceTypeFilterMap(NewInstanceType(ctx, i, p.region,
-			nodeClass.Spec.BlockDeviceMappings, nodeClass.Spec.InstanceStorePolicy,
-			kc.MaxPods, kc.PodsPerCore, kc.KubeReserved, kc.SystemReserved, kc.EvictionHard, kc.EvictionSoft,
-			amiFamily, p.createOfferings(ctx, i, allZones, p.instanceTypeOfferings[aws.StringValue(i.InstanceType)], nodeClass.Status.Subnets),
-		), nodeClass)
+		zoneData := lo.Map(allZones.UnsortedList(), func(zoneName string, _ int) ZoneData {
+			if !p.instanceTypesOfferings[aws.StringValue(i.InstanceType)].Has(zoneName) || !subnetZones.Has(zoneName) {
+				return ZoneData{
+					Name:      zoneName,
+					Available: false,
+				}
+			}
+			return ZoneData{
+				Name:      zoneName,
+				ID:        subnetZoneToID[zoneName],
+				Available: true,
+			}
+		})
+
+		it := p.instanceTypesResolver.Resolve(ctx, i, zoneData, nodeClass)
+		if it == nil {
+			return nil, false
+		}
```

Comment:
Are there any circumstances where this result can be nil? I believe the answer should be no. If we want to include a check, it should be a panic IMO and then we can also change this filter map to a map.